File: leetcode/1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        char_count = 0
        count = 0
        for i in range(len(s)):
            char_count = char_count + 1 if s[i] == "L" else char_count - 1
            if char_count == 0:
                count += 1 
        return count

File: leetcode/1221.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function (s) {
  let count = 0;
  let countSubs = 0;
  for (let i = 0; i < s.length; i++) {
    s[i] === "L" ? count++ : count--;
    // countSubs = (count === 0 ? countSubs + 1 : countSubs)
    count === 0 ? countSubs++ : countSubs;
    // if (count === 0) countSubs++;
  }
  return countSubs;
};
